python/paddle/nn/functional/vision.py

```diff
@@ -102,29 +106,48 @@ def affine_grid(theta, out_shape, align_corners=True, name=None):
             "use_cudnn",
             use_cudnn,
         )
-
-    helper = LayerHelper('affine_grid')
-    check_variable_and_dtype(
-        theta, 'theta', ['float32', 'float64'], 'affine_grid'
-    )
-    out = helper.create_variable_for_type_inference(theta.dtype)
-    ipts = {'Theta': theta}
-    attrs = {"align_corners": align_corners, "use_cudnn": use_cudnn}
-    if isinstance(out_shape, Variable):
-        ipts['OutputShape'] = out_shape
+    elif in_pir_mode():
+        theta = theta._use_gpudnn(use_cudnn)
+        if isinstance(out_shape, (list, tuple)):
+            if paddle.utils._contain_var(out_shape):
+                out_shape = paddle.utils.get_int_tensor_list(out_shape)
+                return _C_ops.affine_grid(
+                    theta,
+                    out_shape,
+                    align_corners,
+                    use_cudnn,
+                )
+        elif isinstance(out_shape, paddle.pir.Value):
+            out_shape.stop_gradient = True
+            return _C_ops.affine_grid(
+                theta,
+                out_shape,
+                align_corners,
+                use_cudnn,
+            )
```

Comment:
这样修改请问会有问题吗？
```suggestion
    elif in_pir_mode():
          return _C_ops.affine_grid(
              theta,
              out_shape,
              align_corners,
          )
```

_C_ops.affine_grid 只有三个输入（即没有 use_cudnn 输入），可以从 `paddle/phi/api/yaml/ops.yaml` 里得知：
```bash
- op : affine_grid
  args : (Tensor input, IntArray output_shape={}, bool align_corners=true)
  output : Tensor(output)
  infer_meta :
    func : AffineGridInferMeta
    param : [input, output_shape, align_corners]
  kernel :
    func : affine_grid
    param : [input, output_shape, align_corners]
    data_type : input
  backward : affine_grid_grad
```

Reply:
收到